base/util/PKICertImport.bash

```diff
@@ -0,0 +1,333 @@
+
+    # Parse the command line arguments and set the appropriate global
+    # variables. Return status of non-zero indicates a fatal error; help
+    # should be displayed. Return status of zero indicates no error and help
+    # should not be displayed.
+    function _parse_args() {
+        # Use a read-and-shift approach to parse both "<option>" and
+        # "<option> <value>" forms.
+        while (( $# > 0 )); do
+            local arg="$1"
+            shift
+
+            # Sorted alphabetically by short option.
+            if [ "x$arg" == "x--ascii" ] || [ "x$arg" == "x-a" ]; then
+                # If specified, the -a flag is passed when the certificate is
+                # imported.
+                CERT_ASCII="true"
+            elif [ "x$arg" == "x--database" ] || [ "x$arg" == "x-d" ]; then
+                # Always required; path to the original NSS DB. Note that this
+                # differs from certutil in that we detect the NSSDB type here,
+                # versus taking a prefix:path combination.
+                NSSDB="$1"
+
+                if [ -e "$NSSDB/cert8.db" ] && [ ! -e "$NSSDB/cert9.db" ]; then
+                    NSSDB_TYPE="dbm:"
+                elif [ ! -e "$NSSDB/cert8.db" ] && [ -e "$NSSDB/cert9.db" ]; then
+                    NSSDB_TYPE="sql:"
+                else
+                    __e "Unknown NSS DB type for directory: $NSSDB"
+                    __e "Please ensure only one of cert8.db or cert9.db exist"
+                    return 1
+                fi
+
+                shift
+            elif [ "x$arg" == "x--password" ] || [ "x$arg" == "x-f" ]; then
+                # If specified, path to a file containing the NSS DB password.
+                NSSDB_PASSWORD="$1"
+                shift
+            elif [ "x$arg" == "x--hsm" ] || [ "x$arg" == "x-h" ]; then
+                # If specified, name of the HSM Token.
+                HSM_TOKEN="$1"
+                shift
+            elif [ "x$arg" == "x--certificate" ] || [ "x$arg" == "x-i" ]; then
+                # Always required; path to the original certificate.
+                CERT_PATH="$1"
+                shift
+            elif [ "x$arg" == "x--nickname" ] || [ "x$arg" == "x-n" ]; then
+                # Always required; nickname for the certificate.
+                CERT_NICKNAME="$1"
+                shift
+            elif [ "x$arg" == "x--trust" ] || [ "x$arg" == "x-t" ]; then
+                # Always required; certificate trust flags.
+                CERT_TRUST="$1"
+                shift
+            elif [ "x$arg" == "x--usage" ] || [ "x$arg" == "x-u" ]; then
+                # Always required; certificate usage flags.
+                CERT_USAGE="$1"
+                shift
+            else
+                # We print help whenever the return code is 1, so we don't
+                # need to explicitly parse a --help flag, but we will get
+                # an extraneous but harmless unknown argument message.
```

Comment:
@ladycfu Do you want me to detect `--help` as an argument directly? I didn't as `--hsm` has the `-h` short flag (to keep syntax like `certutil`). Plus it'll happen anyways (printing help), without needing the additional branch...